### ISSUE TITLE
Feed is not actually updated after adding or removing sources until toggling news switch

### ISSUE THREAD

**Issue by Uni-verse** [brave/brave-browser#37628]
## Description 
Feed is not actually updated after adding or removing sources until toggling news switch

**Regression in 1.65.x**

## Steps to reproduce 

1.  Fresh profile
2. Enable news
3. Remove Top Sources
4. Add any popular sources, confirm in feed (tap on Load new content)
5. Now, open news settings, remove sources from Following , confirm in feed

## Actual result 
Feed is not updated after adding or removing sources

https://github.com/brave/brave-browser/assets/17885425/a8d2b91e-730b-4c64-b7cd-1fb387e7033c

## Expected result
Should update feed

## Issue reproduces how often 
Easily

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? No
- Can you reproduce this issue with the current Play Store Beta version? Yes
- Can you reproduce this issue with the current Play Store Nightly version? TBD

## Device details
   - Install type (ARM, x86): ARM
   - Device type (Phone, Tablet, Phablet): Samsung GS 21
   - Android version: 13
   
## Brave version
1.65.113

**Comment by kjozwiak** [comment]
As discussed via https://bravesoftware.slack.com/archives/C03MMJ4CHEZ/p1713293056106039, we should try getting this resolved for the next minor `1.65.x` release. However, we won't block for the initial `1.65.x` release. CCing @fallaciousreasoning

**Comment by hffvld** [comment]
Verified on `Pixel 7` using version(s):
```
Device/OS: Pixel 7 / panther_beta-user 14 AP31.240426.023.B4 release-keys
Brave build: 1.67.100
Chromium: 125.0.6422.112 (Official Build) beta (64-bit) 
```
---------------------------

**STEPS:**
1. Follow the steps from https://github.com/brave/brave-browser/issues/37628#issue-2246668669
2. *Verify*

**ACTUAL RESULTS:**
- Verified that the `News` feed is updated without Brave relaunch or enable/disable the `News` toggle switch when adding/removing news sources. 

---------------------------

https://github.com/brave/brave-browser/assets/128532965/ecfe9066-c0b3-437c-8e79-d2100e47a102
